src/components/Frequency/Frequency.tsx

```diff
@@ -0,0 +1,176 @@
+import React, { useState, useMemo, useCallback, useRef } from "react";
+import { nanoid } from "nanoid";
+import Field from "../internal/Field";
+import Grid from "../Grid";
+import InternalInput from "../internal/InternalInput";
+import InternalRadioGroup from "../internal/InternalRadioGroup";
+import InternalSelect from "../internal/InternalSelect";
+import { InternalFrequencyProps } from "./types";
+import { useMergedProps } from "../../hooks/useMergedProps";
+import { defaultFrequencyProps } from "./defaultFrequencyProps";
+import mergeRefs from "../../utils/mergeRefs";
+import { ALL_FREQUENCY_OPTIONS, FrequencyValue } from "../../values";
+
+function Frequency(props: InternalFrequencyProps) {
+  const mergedProps = useMergedProps(props, defaultFrequencyProps);
+  const {
+    mode,
+    label,
+    annually,
+    innerRef = null,
+    error,
+    quarterly,
+    monthly,
+    fortnightly,
+    weekly,
+    amountPrefix,
+    amountPlaceholder,
+    selectPlaceholder,
+    amountMaxLength,
+    helpText,
+    disabled,
+    optional,
+    testId,
+    onFocus,
+    onBlur,
+    onChange,
+    onMouseDown,
+  } = mergedProps;
+  const [labelId] = useState(() => `frequency-label-${nanoid()}`);
+  const [auxId] = useState(() => `frequency-aux-${nanoid()}`);
+
+  const frequencyOptions = useMemo(() => {
+    const frequencyPropsMap = {
+      annually,
+      quarterly,
+      monthly,
+      fortnightly,
+      weekly,
+    };
+    return ALL_FREQUENCY_OPTIONS.filter(
+      (option) => frequencyPropsMap[option.value]
+    );
+  }, [annually, quarterly, monthly, fortnightly, weekly]);
+
+  const [internalValue, setInternalValue] = useState<FrequencyValue>({
+    amount: "",
+    frequency: "",
+  });
+
+  const value = props.value ?? internalValue;
+
+  const onFrequencyChange: React.ChangeEventHandler<
+    HTMLInputElement | HTMLSelectElement
+  > = (event) => {
+    const { name: key, value: updatedValue } = event.target;
+    if (key !== "amount" && key !== "frequency") {
+      return;
+    }
+    onChange?.({
+      ...value,
+      [key]: updatedValue,
+    });
+
+    setInternalValue((val) => ({
+      ...val,
+      [key]: updatedValue,
+    }));
+  };
+
+  const frequencyRef = useRef<HTMLDivElement>(null);
+
+  const fieldErrors = Object.values(error ?? {});
+  const hasErrors = !!error;
+
+  const onFrequencyBlur: React.FocusEventHandler<
+    HTMLInputElement | HTMLSelectElement
+  > = useCallback(
+    (event) => {
+      if (
+        !frequencyRef.current?.contains(event.relatedTarget as HTMLInputElement)
```

Comment:
Wondering why  we need to cast here? can't we have proper types for event ?

Reply:
It does have proper types for event, however the types for `relatedTarget` are a bit complex and give a number of typescript errors. 

It's just easier to cast in this case for `event.relatedTarget`.